src/preact/services/platform.js

```diff
@@ -0,0 +1,194 @@
+class PlatformService {
```

Comment:
I'm assuming this is a straightforward copy of the `platform-impl.js` code?

Reply:
Yes, a straighforward copy, very few mods.